packages/compat/src/import-parser.ts

```diff
@@ -61,7 +68,20 @@ export default class ImportParser extends Plugin {
     return f.parserOpts;
   }
 
-  build() {
+  async ensureParserSetup() {
+    if (this.babelMajorVersion === 7) {
+      const { parseSync } = await BabelCore7;
```

Comment:
this could be `await import('@babel/core')` and then it won't load unless needed.